src/overlays/actors/ovl_Obj_Um/z_obj_um.c

```diff
@@ -46,131 +73,1951 @@ const ActorInit Obj_Um_InitVars = {
+
+    ObjUm_SetPlayerPosition(this, play);
+    this->flags |= OBJ_UM_FLAG_0004;
+    player->stateFlags1 |= PLAYER_STATE1_20;
+    if (ActorCutscene_GetCanPlayNext(this->dyna.actor.cutscene)) {
+        ActorCutscene_StartAndSetUnkLinkFields(this->dyna.actor.cutscene, &this->dyna.actor);
+        this->lastTime = gSaveContext.save.time;
+        ObjUm_SetupAction(this, func_80B7A394);
+    } else {
+        ActorCutscene_SetIntentToPlay(this->dyna.actor.cutscene);
+    }
+}
+
+void ObjUm_RunMinigame(ObjUm* this, PlayState* play) {
+    ObjUm_SetPlayerPosition(this, play);
+    ObjUm_RotatePlayer(this, play, 0x7FFF);
+    this->wheelRot += 2000;
+    this->flags |= OBJ_UM_FLAG_0010;
+    ObjUm_UpdateAnim(this, play, OBJ_UM_ANIM_GALLOP);
+
+    switch (ObjUm_UpdatePath(this, play)) {
+        case OBJUM_PATH_STATE_1:
+        case OBJUM_PATH_STATE_FINISH:
+            gSaveContext.seqIndex = 0xFF;
+            gSaveContext.save.weekEventReg[31] &= (u8)~0x80;
+            gSaveContext.nightSeqIndex = 0xFF;
+
+            if (!(gSaveContext.save.weekEventReg[52] & 1) && !(gSaveContext.save.weekEventReg[52] & 2)) {
+                if (this->arePotsBroken == false) {
+                    play->nextEntranceIndex = 0x3E60;
+                    play->unk_1887F = 0x40;
+                    gSaveContext.nextTransition = 3;
+                    play->sceneLoadFlag = 0x14;
+                    gSaveContext.save.weekEventReg[52] |= 1;
+                    gSaveContext.save.weekEventReg[52] &= (u8)~2;
+                } else {
+                    play->nextEntranceIndex = 0x6480;
+                    play->unk_1887F = 0x40;
+                    gSaveContext.nextTransition = 3;
+                    play->sceneLoadFlag = 0x14;
+                    gSaveContext.save.weekEventReg[52] |= 2;
+                    gSaveContext.save.weekEventReg[52] &= (u8)~1;
+                }
+            }
+            break;
+
+        default:
+            Actor_MoveWithGravity(&this->dyna.actor);
+            func_80B78DF0(this, play);
+            break;
+    }
+}
+
+void func_80B7A614(ObjUm* this, PlayState* play) {
+    s32 pad;
+
+    ObjUm_SetPlayerPosition(this, play);
+    ObjUm_RotatePlayer(this, play, 0x7FFF);
+    this->wheelRot += 2000;
+    this->flags |= OBJ_UM_FLAG_0010;
+    this->flags |= OBJ_UM_FLAG_PLAYING_MINIGAME;
+    ObjUm_UpdateAnim(this, play, OBJ_UM_ANIM_GALLOP);
+
+    if (ObjUm_UpdatePath(this, play) == OBJUM_PATH_STATE_3 && this->unk_4DC == 0) {
+        this->unk_4DC = 1;
+    } else if (this->unk_4DC > 0) {
+        if (this->unk_4DC == 1) {
+            s32 i;
+
+            this->arePotsBroken = true;
+
+            for (i = 0; i < ARRAY_COUNT(this->potsLife); i++) {
+                if (this->potsLife[i] != 1) {
+                    this->arePotsBroken = false;
+                    break;
+                }
+            }
+
+            this->flags |= OBJ_UM_FLAG_MINIGAME_FINISHED;
+
+        dummy_label:;
```

Comment:
I wonder if you can toss that pad from above somewhere in one of these scopes to fix this dummy label match. Do you have a decomp.me handy?

Reply:
https://decomp.me/scratch/INtIN

Reply:
Seems fine to just remove it? https://decomp.me/scratch/ZR2LY